File: 参考/comp9041-Q/cs9041/lab06/all_whales.pl
#!/usr/bin/perl -w
while ($input = <STDIN>){
	chomp $input;
	$input =~ /^(\d+)\s*(.*)$/ or die;
	$tmp = $2;
	$num = $1;
	$tmp =~ tr/[A-Z]/[a-z]/;
	$tmp =~ tr/' '/' '/s;
	$tmp =~ s/ $//g;
	$tmp =~ s/s$//i;
	


	if (not $data{$tmp}){
		$data{$tmp} = "$num";
	}
	else{
		$data{$tmp} = "$data{$tmp},$num";
	}
}


@key = keys %data;
@key = sort(@key);
foreach $key (@key){
	@arr = split(',',$data{$key});
	$sum=eval(join("+",@arr));
	$pod=$#arr+1;
	print "$key observations: $pod pods, $sum individuals\n";
}

File: others/comp9041-Q/cs9041/lab06/all_whales.py
#!/usr/bin/python3
from collections import defaultdict
import re,sys
a = defaultdict(list)
b= []
while True:
	s = sys.stdin.readline()
	if not s:
		break
	s=s.strip()
	match = re.match(r'^(\d+) (.*)$',s)
	try:
		num = match.group(1)
		# print(num)
		name = match.group(2)
		name = name.lower()
		if name[-1] == 's':
			name = name[:-1]
		name = ' '.join(name.split())
	except:
		pass
	a[name].append(int(num))
# 	b.append(name)
# b.sort()
for key in sorted(a):
	print("{} observations: {} pods, {} individuals"\
		.format(key,len(a[key]),sum(a[key])))